Modules/cjkcodecs/cjkcodecs.h

```diff
@@ -249,53 +282,70 @@ getmultibytecodec(void)
 }
 
 static PyObject *
-getcodec(PyObject *self, PyObject *encoding)
+_getcodec(const MultibyteCodec *codec)
 {
-    PyObject *codecobj, *r, *cofunc;
-    const MultibyteCodec *codec;
-    const char *enc;
-
-    if (!PyUnicode_Check(encoding)) {
-        PyErr_SetString(PyExc_TypeError,
-                        "encoding name must be a string.");
+    PyObject *cofunc = getmultibytecodec();
+    if (cofunc == NULL) {
         return NULL;
     }
-    enc = PyUnicode_AsUTF8(encoding);
-    if (enc == NULL)
-        return NULL;
 
-    cofunc = getmultibytecodec();
-    if (cofunc == NULL)
+    PyObject *codecobj = PyCapsule_New((void *)codec,
+                                       PyMultibyteCodec_CAPSULE_NAME,
+                                       NULL);
+    if (codecobj == NULL) {
+        Py_DECREF(cofunc);
         return NULL;
+    }
 
-    for (codec = codec_list; codec->encoding[0]; codec++)
-        if (strcmp(codec->encoding, enc) == 0)
-            break;
+    PyObject *res = PyObject_CallOneArg(cofunc, codecobj);
```

Comment:
Please check that if PyObject_CallOneArg returns NULL :)

Reply:
`res` is propagated to the caller ;)